addons/subscription/models/subscription.py

```diff
@@ -0,0 +1,122 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+# TODO:
+#   Error treatment: exception, request, ... -> send request to user_id
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+class SubscriptionDocument(models.Model):
+    _name = "subscription.document"
+    _description = "Subscription Document"
+
+    name = fields.Char(required=True)
+    active = fields.Boolean(help="If the active field is set to False, it will allow you to hide the subscription document without removing it.", default=True)
+    model_id = fields.Many2one('ir.model', string="Object", required=True)
+    field_ids = fields.One2many('subscription.document.fields', 'document_id', string='Fields', copy=True)
+
+class SubscriptionDocumentFields(models.Model):
+    _name = "subscription.document.fields"
+    _description = "Subscription Document Fields"
+    _rec_name = 'field_id'
+
+    field_id = fields.Many2one('ir.model.fields', domain="[('model_id', '=', parent.model_id)]", required=True)
+    value = fields.Selection([('false', 'False'), ('date', 'Current Date')], string='Default Value', help="Default value is considered for field when new document is generated.")
+    document_id = fields.Many2one('subscription.document', string='Subscription Document', ondelete='cascade')
+
+def _get_document_types(self):
```

Comment:
why this method is outside of class ? if need, can not declare after import statement ?